.github/workflows/bump.yml

```diff
@@ -0,0 +1,187 @@
+          else
+            CHANGED_FILES=$(git diff --name-only ${{ github.event.before }} ${{ github.event.after }})
+          fi
+          echo "CHANGED_FILES<<EOF" >> $GITHUB_ENV
+          echo "$CHANGED_FILES" >> $GITHUB_ENV
+          echo "EOF" >> $GITHUB_ENV
+
+      - name: Set matrix
+        id: set-matrix
+        run: |
+          DOCS=()
+
+          # Check for changes in API doc folders
+          if echo "$CHANGED_FILES" | grep -q "^admin/"; then
+            DOCS+=("admin")
+          fi
+          if echo "$CHANGED_FILES" | grep -q "^cloud-controlplane/"; then
+            DOCS+=("cloud-controlplane")
+          fi
+          if echo "$CHANGED_FILES" | grep -q "^cloud-dataplane/"; then
+            DOCS+=("cloud-dataplane")
+          fi
+          if echo "$CHANGED_FILES" | grep -q "^schema-registry/"; then
+            DOCS+=("schema-registry")
+          fi
+          if echo "$CHANGED_FILES" | grep -q "^http-proxy/"; then
+            DOCS+=("http-proxy")
+          fi
+          # Check for changes in shared resources that might affect all docs
+          if echo "$CHANGED_FILES" | grep -q "^shared/"; then
+            if [ ${#DOCS[@]} -eq 0 ]; then
+              # If only shared files changed and no specific doc files, include all docs
+              DOCS+=("admin" "cloud-controlplane" "cloud-dataplane" "schema-registry" "http-proxy")
+            fi
+          fi
+
+          # If no files changed in any monitored directories, abort the workflow
+          if [ ${#DOCS[@]} -eq 0 ]; then
+            echo "No relevant files were changed. Exiting workflow."
+            echo "matrix={\"doc_id\":[]}" >> $GITHUB_OUTPUT
+            exit 0
+          fi
+
+          # Convert bash array to JSON array for GitHub Actions matrix
+          JSON_ARRAY=$(printf '"%s",' "${DOCS[@]}" | sed 's/,$//')
+          JSON_MATRIX="{\"doc_id\":[$JSON_ARRAY]}"
+          echo "matrix=$JSON_MATRIX" >> $GITHUB_OUTPUT
+          echo "Created matrix: $JSON_MATRIX"
+
+  deploy-doc:
+    if: ${{ github.event_name == 'push' && fromJson(needs.determine-doc-ids.outputs.matrix).doc_id[0] != null }}
+    needs: determine-doc-ids
+    name: Deploy API documentation on Bump.sh
+    runs-on: ubuntu-latest
+    strategy:
+      matrix: ${{fromJson(needs.determine-doc-ids.outputs.matrix)}}
+      fail-fast: false
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Determine file format
+        id: format
+        run: |
+          if [ -f "${{ matrix.doc_id }}/${{ matrix.doc_id }}.yaml" ]; then
+            echo "file_path=${{ matrix.doc_id }}/${{ matrix.doc_id }}.yaml" >> $GITHUB_OUTPUT
+          elif [ -f "${{ matrix.doc_id }}/${{ matrix.doc_id }}.json" ]; then
+            echo "file_path=${{ matrix.doc_id }}/${{ matrix.doc_id }}.json" >> $GITHUB_OUTPUT
+          elif [ -f "${{ matrix.doc_id }}/${{ matrix.doc_id }}-api.json" ]; then
```

Comment:
Is the check for a `-api.json` just in case? (No need for the yaml equivalent?)

Reply:
I'll remove that check. It was from before I updated the filenames 😄